flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSink.java

```diff
@@ -105,7 +153,18 @@ public void invoke(IN value) {
 		try {
 			byte[] msg = schema.serialize(value);
 
-			channel.basicPublish("", queueName, null, msg);
+			if (messageCompute == null) {
+				channel.basicPublish("", queueName, null, msg);
+			} else {
+				String rk = messageCompute.computeRoutingKey(value);
+				String exchange = messageCompute.computeExchange(value);
+				channel.basicPublish((exchange != null) ? exchange : "",
```

Comment:
I think it is better to assume that `computeRoutingKey` and `computeExchange` must return a non-null value. It would be consistent with the `basicPublish` method which also rejects nulls.

Reply:
ping